server/schedule/operator_controller.go

```diff
@@ -264,15 +265,18 @@ func (oc *OperatorController) AddWaitingOperator(ops ...*operator.Operator) bool
 			oc.buryOperator(op)
 		}
 		oc.Unlock()
+		log.Debug("exceed_max return false", zap.Uint64("waiting", oc.wopStatus.ops[desc]), zap.String("desc", desc), zap.Uint64("max", oc.cluster.GetSchedulerMaxWaitingOperator()))
 		return false
 	}
 	oc.wop.PutOperator(op)
 	operatorWaitCounter.WithLabelValues(op.Desc(), "put").Inc()
-	// This step is especially for the merge operation.
+	oc.wopStatus.ops[desc]++
 	if len(ops) > 1 {
-		oc.wop.PutOperator(ops[1])
+		for _, op := range ops[1:] {
+			oc.wop.PutOperator(op)
+			oc.wopStatus.ops[desc]++
```

Comment:
After the change, the odd number may let the merge operator only push one operator to the waiting operator and the other one exceed the max limit. please add a unit test to make sure.

Reply:
The check `oc.wopStatus.ops[desc] >= oc.cluster.GetSchedulerMaxWaitingOperator()` has been done above.